### ISSUE TITLE
Markdown exercise does not pass tests by default

### ISSUE THREAD

**Issue by enderger** [exercism/bash#461]
In the Markdown exercise, the default code does not pass test 14.
![](https://user-images.githubusercontent.com/36283171/100528108-e100ae00-318d-11eb-86e6-f8927083a62d.png)
This is in conflict with the instructions, which claim
> The markdown exercise is a refactoring exercise. There is code that parses a given string with Markdown syntax and returns the associated HTML for that string. Even though this code is confusingly written and hard to follow, _somehow it works and all the tests are passing_!

**Comment by glennj** [comment]
I'm not seeing any failures:
```bash
$ env BATS_RUN_SKIPPED=true bats markdown_test.sh
 ✓ parses normal text as a paragraph
 ✓ parsing italics
 ✓ parsing multiple italics
 ✓ parsing bold text
 ✓ mixed normal, italics and bold text
 ✓ with h1 header level
 ✓ with h2 header level
 ✓ with h6 header level
 ✓ unordered lists
 ✓ With a little bit of everything
 ✓ with markdown symbols in the header text that should not be interpreted
 ✓ with markdown symbols in the list item text that should not be interpreted
 ✓ with markdown symbols in the paragraph text that should not be interpreted
 ✓ unordered lists close properly with preceding and following lines

14 tests, 0 failures
```
What platform are you on? What versions of bash, grep, expr?
What do you see running that test manually:
```bash
bash -x markdown.sh <(echo 'This is a paragraph with # and * in the text')
```

**Comment by enderger** [comment]
System: NixOS Unstable Channel
Bash Version: GNU 4.4.23
Grep Version: GNU 3.5
Expr Version: UUtils 0.0.1
Command Output (removed -x since it is an output equality error): `<h1>his is a paragraph with # and * in the text</h1>`

**Comment by glennj** [comment]
The trace output may prove helpful in determining where your system goes off course. I see
```bash
$ bash -x markdown.sh <(echo 'This is a paragraph with # and * in the text')
+ IFS=
+ read -r line
+ true
+ orig='This is a paragraph with # and * in the text'
+ [[ This is a paragraph with # and * in the text =~ ^(.+)__(.*) ]]
+ '[' 'This is a paragraph with # and * in the text' '!=' 'This is a paragraph with # and * in the text' ']'
+ break
+ echo 'This is a paragraph with # and * in the text'
+ grep '^\*'
+ '[' 1 -eq 0 ']'
+ '[' X = Xyes ']'
++ expr 'This is a paragraph with # and * in the text' : '#\{1,\}'
+ n=0
+ '[' 0 -gt 0 ']'
+ grep '_..*_'
+ h='<p>This is a paragraph with # and * in the text</p>'
+ IFS=
+ read -r line
+ '[' X = Xyes ']'
+ echo '<p>This is a paragraph with # and * in the text</p>'
<p>This is a paragraph with # and * in the text</p>
```
The key thing here to determine if the line is a header is this test
```bash
expr 'This is a paragraph with # and * in the text' : '#\{1,\}'
```
outputting non-zero

**Comment by glennj** [comment]
@enderger are you OK now? Can I close this issue?

**Comment by enderger** [comment]
I think the issue is with the UUtils expr as well, as it appears to be truncating the 'T' from "this".
